booster/library/Booster/Pattern/Match.hs

```diff
@@ -212,8 +214,7 @@ match1 _       t1@DomainValue{}                           t2@KList{}
 match1 _       t1@DomainValue{}                           t2@KSet{}                                  = failWith $ DifferentSymbols t1 t2
 match1 _       t1@DomainValue{}                           t2@ConsApplication{}                       = failWith $ DifferentSymbols t1 t2
 match1 _       t1@DomainValue{}                           t2@FunctionApplication{}                   = addIndeterminate t1 t2
-match1 Rewrite t1@DomainValue{}                           (Var var2)                                 = matchVar Rewrite var2 t1
-match1 Eval    t1@DomainValue{}                           t2@Var{}                                   = addIndeterminate t1 t2
+match1 _       t1@DomainValue{}                           (Var t2)                                   = failWith $ SubjectVariableMatch t1 t2
```

Comment:
As also discussed on slack: 
This needs to remain `addIndeterminate` (and we have to make sure we have a test for it)
See https://github.com/runtimeverification/hs-backend-booster/issues/231